src/main/java/fr/jmmc/oiexplorer/core/model/OIFitsCollectionManager.java

```diff
@@ -606,11 +596,54 @@ public OIFitsFile removeOIFitsFile(final OIFitsFile oiFitsFile) {
                 }
             }
 
-            fireOIFitsCollectionChanged();
+            if (fireEvent) {
+                fireOIFitsCollectionChanged();
+            }
         }
         return previous;
     }
 
+    /**
+     * alias of removeOIFitsFile with fireEvent set to true. Remove the OIDataFile given its corresponding OIFits
+     * structure (filePath matching)
+     *
+     * @param oiFitsFile OIFits structure
+     * @return removed OIDataFile or null if not found
+     */
+    public OIFitsFile removeOIFitsFile(final OIFitsFile oiFitsFile) {
+        return removeOIFitsFile(oiFitsFile, true);
+    }
+
+    /**
+     * removes from collection every OIFits file of the given list. Then fires an event.
+     *
+     * @param listOIfitsfiles list of OIFits files to remove
+     * @return list of deleted OIFits file or null elements (null when the file was not found)
+     */
+    public List<OIFitsFile> removeListOIFitsFile(final List<OIFitsFile> listOIfitsfiles) {
+        final List<OIFitsFile> listPrevious = new ArrayList<>(listOIfitsfiles.size());
+
+        for (OIFitsFile oiFitsFile : listOIfitsfiles) {
+            listPrevious.add(removeOIFitsFile(oiFitsFile, false));
```

Comment:
why add null in list of removed files ? it means the caller will have to deal with nulls ... (complicated ?)